framework/xds_k8s_testcase.py

```diff
@@ -239,9 +239,11 @@ def setUpClass(cls):
         cls.check_local_certs = _CHECK_LOCAL_CERTS.value
 
         # Resource managers
-        cls.k8s_api_manager = k8s.KubernetesApiManager(
-            xds_k8s_flags.KUBE_CONTEXT.value
-        )
+        if not hasattr(cls, "k8s_api_manager"):
```

Comment:
This just seems wrong to me. We added it because `CloudRunXdsTestCase` doesn't use kubernetes. But it extends `CloudRunXdsKubernetesTestCase`, which extends this class, `XdsKubernetesBaseTestCase`. 

I mean it only makes sense to have `k8s_api_manager` in `XdsKubernetesBaseTestCase`, which implies kubernetes will be used.

Reply:
Ideally we'd want to break up the classes so that there's a class that parses all the flags except kubernetes. Then have `XdsKubernetesBaseTestCase` that extends it and adds k8s-specific features. And have `CloudRunXdsTestCase` (name is up for discussion) that extends it - and doesn't use k8s at all.

However, I'll be ok if we made this change in a separate PR and only added todos in this one.
Keep in mind that the scripts that initiate test runs also assume we run on k8s - perform cluster auth, etc.